src/tracy_impl.rs

```diff
@@ -1,11 +1,21 @@
 #[macro_export]
 macro_rules! scope {
     ($name:expr) => {
-        // Note: callstack_depth is 0 since this has significant overhead
-        let _tracy_span = $crate::tracy_client::Span::new($name, "", file!(), line!(), 0);
+        struct S;
+        let type_name = core::any::type_name::<S>();
+        let function_name = &type_name[..type_name.len() - 3];
+        let _tracy_span = $crate::tracy_client::Client::running()
```

Comment:
Is there a reason this can't be the span!() macro similar to what the procmacro emits?

If we can't use the span!() macro directly, is it possible to use the span_location!() macro?

BTW I'm not sure this would compile if multiple scopes were in the same function since it looks like the struct S would be defined multiple times.

Reply:
> BTW I'm not sure this would compile if multiple scopes were in the same function since it looks like the struct S would be defined multiple times.

Thanks for catching this, I fixed it in the latest commit.

Reply:
As an optimization, the macro could first try to match on name being a `literal`. Thoughts on this?

Reply:
I can't think of a reason not to. Worst case, it doesn't work and we change it to assume it's not literal.

(BTW sorry I've been a bit slow to respond, I've been very busy lately. Thanks for taking the time to PR this!)

Reply:
I implemented this and I was testing things to make sure the non-literal case still worked and I noticed that `puffin` seems to require a `&'static str` and `tracing` needs it to be a constant. So I guess some backends already won't let this be any `&str`?
![image](https://user-images.githubusercontent.com/2002109/168957498-9ede6276-2496-4abd-b62d-a107b038eda4.png)
![image](https://user-images.githubusercontent.com/2002109/168957424-126396b7-12b7-4dbf-be2c-820ba89afe85.png)

Reply:
Maybe this would justify just using the macro from `tracy` directly in all cases? I'm not sure.

Reply:
yeah it might, I'm not sure either. Happy to table it to some time in the future. Essentially a decision of what our API is precisely. There is a backend that does nothing, that essentially exists to have CI check compatibility with this lib, currently that is the best, most "authoritative" specification for the API.